### ISSUE TITLE
EXC_BAD_ACCESS when making request using concurrency in iOS 13, especially building on Xcode 14.1

### ISSUE THREAD

**Issue by hellojunha** [Alamofire/Alamofire#3663]
> ℹ Please fill out this template when filing an issue.
> All lines beginning with an ℹ symbol instruct you with what info we expect. You can delete those lines once you've filled in the info.
>
> Per our [*CONTRIBUTING guidelines](https://github.com/Alamofire/Alamofire/blob/master/CONTRIBUTING.md), we use GitHub for
> bugs and feature requests, not general support. Other issues should be opened on Stack Overflow with the tag `alamofire`.
>
> Please remove this line and everything above it before submitting.

* [x] I've read, understood, and done my best to follow the [*CONTRIBUTING guidelines](https://github.com/Alamofire/Alamofire/blob/master/CONTRIBUTING.md).

## What did you do?

Created a new project on Xcode 14.1 with simple network request.

No error is occurred when I make a request with completionHandler, but error occurs when I do same thing using concurrency (async/await).

## What did you expect to happen?

Successful network request with no error

## What happened instead?

EXC_BAD_ACCESS occurred from `closure #1 in closure #3 in closure #1 in DataRequest._response<τ_0_0>(queue:responseSerializer:completionHandler:) at /Pods/Alamofire/Source/ResponseSerialization.swift:244`

But it does not happens when I build on Xcode 14.0.1, but only on Xcode 14.1
Also, not happening on iOS 14, only on iOS 13

## Alamofire Environment

**Alamofire Version:** 5.6.2
**Dependency Manager:** CocoaPods 1.11.3
**Xcode Version:** 14.1
**Swift Version:** 5.7.1
**Platform(s) Running Alamofire:** iOS 13
**macOS Version Running Xcode:** 12.5.1

## Demo Project

ℹ Please link to or upload a project we can download that reproduces the issue.

[demo2.zip](https://github.com/Alamofire/Alamofire/files/10039151/demo2.zip)

**Comment by jshier** [comment]
To investigate this we need the full crash stack from your report. Most likely this is a concurrency runtime issue but we'll need the report to be sure.

**Comment by hellojunha** [comment]
Following is stack trace

```
(lldb) frame info
frame #0: 0x000000010b875451 libobjc.A.dylib`objc_destructInstance + 44
(lldb) bt
* thread #4, queue = 'org.alamofire.concurrencySingleEventQueue', stop reason = EXC_BAD_ACCESS (code=2, address=0x1000000050)
  * frame #0: 0x000000010b875451 libobjc.A.dylib`objc_destructInstance + 44
    frame #1: 0x000000010bbbf198 libswiftCore.dylib`swift_deallocClassInstance + 24
    frame #2: 0x000000010bbbf0d0 libswiftCore.dylib`_swift_release_dealloc + 16
    frame #3: 0x000000010a9f448b Alamofire`___lldb_unnamed_symbol36697 + 155
    frame #4: 0x000000010bbbf0d0 libswiftCore.dylib`_swift_release_dealloc + 16
    frame #5: 0x000000010aabf948 Alamofire`___lldb_unnamed_symbol36864 + 168
    frame #6: 0x000000010bbbf0d0 libswiftCore.dylib`_swift_release_dealloc + 16
    frame #7: 0x000000010aac026d Alamofire`block_destroy_helper.197 at <compiler-generated>:0
    frame #8: 0x000000010dfed9ba libsystem_blocks.dylib`_Block_release + 101
    frame #9: 0x000000010debbe8e libdispatch.dylib`_dispatch_client_callout + 8
    frame #10: 0x000000010debe7a3 libdispatch.dylib`_dispatch_continuation_pop + 552
    frame #11: 0x000000010debdbbb libdispatch.dylib`_dispatch_async_redirect_invoke + 771
    frame #12: 0x000000010decd399 libdispatch.dylib`_dispatch_root_queue_drain + 351
    frame #13: 0x000000010decdca6 libdispatch.dylib`_dispatch_worker_thread2 + 135
    frame #14: 0x000000010e379c9d libsystem_pthread.dylib`_pthread_wqthread + 256
    frame #15: 0x000000010e378c67 libsystem_pthread.dylib`start_wqthread + 15
```

**Comment by jshier** [comment]
I can't replicate this in the iOS 13 simulator under Xcode 14.1 or 14.2. Are you only seeing this on device? I suggest you try under the recently released Xcode 14.2 RC and see if the behavior has changed again. Unfortunately Swift's concurrency system has shipped a variety of bugs for different Xcode releases, so I'm not sure what Alamofire can do here. 

If this continues to occur, please post a symbolicated crash log, not one from the debugger, and we can investigate further.

**Comment by hellojunha** [comment]
I encountered this problem with two different Macs and both for simulator and real device, so I thought this is not a problem just for me. But fortunately, I am not seeing this problem in recently released Xcode 14.2. I will let you know if I find something new about this issue.
